src/piton/labelers/core.py

```diff
@@ -444,66 +469,78 @@ def label(self, patient: Patient) -> List[Label]:
         outcome_times: List[datetime.datetime] = self.get_outcome_times(patient)
         time_horizon: TimeHorizon = self.get_time_horizon()
 
+        # Get (start, end) of time horizon. If end is None, then it's infinite (set timedelta to max)
+        time_horizon_start: datetime.timedelta = time_horizon.start
+        time_horizon_end: Optional[
+            datetime.timedelta
+        ] = time_horizon.end  # `None` if infinite time horizon
+
         results: List[Label] = []
         curr_outcome_idx: int = 0
         # For each prediction time, check if there is an outcome which occurs within the (start, end)
         # of the time horizon
         for time in self.get_prediction_times(patient):
             while (
-                curr_outcome_idx < len(outcome_times)
-                and outcome_times[curr_outcome_idx] < time + time_horizon.start
+                curr_outcome_idx < len(outcome_times) - 1
+                and outcome_times[curr_outcome_idx] < time + time_horizon_start
             ):
-                # This is the idx in `outcome_times` that corresponds to the first outcome EQUAL or AFTER
-                # the time horizon for this prediction time starts (if one exists)
+                # `curr_outcome_idx` is the idx in `outcome_times` that corresponds to the first
+                # outcome EQUAL or AFTER the time horizon for this prediction time starts (if one exists)
                 curr_outcome_idx += 1
 
             # TRUE if an event occurs within the time horizon
-            is_outcome_occurs_in_time_horizon: bool = curr_outcome_idx < len(
-                outcome_times
-            ) and (
-                time + time_horizon.start
-                <= outcome_times[curr_outcome_idx]
-                <= time + time_horizon.end
+            is_outcome_occurs_in_time_horizon: bool = (
+                (
+                    # ensure there is an outcome
+                    # (needed in case there are 0 outcomes)
+                    curr_outcome_idx
+                    < len(outcome_times)
+                )
+                and (
```

Comment:
It's needed in case there are 0 outcomes and 1+ prediction times -- 

We start with

`curr_outcome_idx = 0`

then

`while (curr_outcome_idx < len(outcome_times) and outcome_times[curr_outcome_idx] < time + time_horizon_start):`

will be false since `len(outcome_times) == 0`

so then if you try to do `outcome_times[curr_outcome_idx]` on line 519, you will get an error since `outcome_times[0]` does not exist